services/src/floodlight-modules/src/main/java/org/openkilda/floodlight/switchmanager/SwitchManager.java

```diff
@@ -534,82 +525,73 @@ public long installOneSwitchFlow(
         return result;
     }
 
-    /**
-     * {@inheritDoc}
-     */
     @Override
-    public long installMeter(final DatapathId dpid, final long bandwidth, final long burstSize, final long meterId)
-            throws SwitchOperationException {
-        long meterCommandXid = 0L;
+    public OFMeterConfig getMeter(DatapathId dpid, long meterId) throws SwitchOperationException {
+        IOFSwitch sw = lookupSwitch(dpid);
+        OFFactory ofFactory = sw.getOFFactory();
 
-        if (meterId == 0) {
-            logger.info("skip installing meter {} on switch {} width bandwidth {}", meterId, dpid, bandwidth);
-            return meterCommandXid;
-        }
+        OFMeterConfigStatsRequest meterRequest = ofFactory.buildMeterConfigStatsRequest()
+                .setMeterId(meterId)
+                .build();
 
-        IOFSwitch sw = lookupSwitch(dpid);
+        try {
+            ListenableFuture<OFMeterConfigStatsReply> future = sw.writeRequest(meterRequest);
+            OFMeterConfigStatsReply reply = future.get(1, TimeUnit.SECONDS);
```

Comment:
1 second could be not enough for heavily loaded switch... with a lot of pkt_out/pkt_in commands. And we should make it using some constant or config value (I would like to see it as a class level constant).

Later we should replace this "sleep" calls with tools proposed in https://github.com/telstra/open-kilda/pull/1601.